src/rust_connection/mod.rs

```diff
@@ -617,21 +653,74 @@ fn write_all_vectored(
         }
     }
     if !fds.is_empty() {
-        write.write_all(&[], fds)?;
+        let count = write_polling_repeat(write_fd, || write.write(&[], &mut fds))?;
+        assert_eq!(count, 0);
     }
     Ok(())
 }
 
+/// Wait for some FD to satisfy the given poll flags
+fn do_poll(fd: RawFd, flags: PollFlags) -> std::io::Result<()> {
+    let mut fds = [PollFd::new(fd, flags)];
+    let res = poll(&mut fds, -1)
+        // errno still contains the error code
+        .map_err(|_| IOError::last_os_error())?;
+
+    // The timeout could not have expired and there was no error, so our one FD must be readable.
+    assert_eq!(res, 1);
+    assert!(
+        fds[0].revents().map(|e| e.contains(flags)).unwrap_or(false),
+        "{:?}", fds[0].revents(),
+    );
```

Comment:
Hrm. This `assert` is kind of incorrect. Socket errors are also a possible reason for wakeups. In this case, the assert would incorrectly trigger.